### PR TITLE
feat(jtk): fix issues get --extended sprint resolution and transitions format

### LINKED ISSUE #303: feat(jtk): fix issues get --extended sprint resolution and transitions format
### 1. `issues get --extended` — Sprint line not resolving
Sprint shows `-` even when the issue has a sprint. The sprint data is present in the custom fields block (`customfield_10020: MON Sprint 70   (Sprint)`) but the dedicated `Sprint:` line doesn't pick it up. Extended spec expects:
```
Sprint: MON Sprint 70 (id: 125, active, 2026-04-10 → 2026-04-24)
```

### 2. `issues get --extended` — Transitions block format
Current output uses a table with a header row:
```
Transitions:
ID | NAME
11 | Backlog
21 | Ready for Development
```
Spec requires indented pipe-delimited lines with no header:
```
Transitions:
  11 | Backlog
  21 | Ready for Development
```

---

## Success criteria

Build and lint before testing:
```
make build-jtk && make lint
```

Use **MON-4970** as the test issue — it has a sprint (`customfield_10020: MON Sprint 70`, sprint id 125) and a full set of transitions.

**1. Sprint line resolves.** Run:
```
bin/jtk issues get MON-4970 --extended
```
The `Sprint:` line must read:
```
Sprint: MON Sprint 70 (id: 125, active, 2026-04-10 → 2026-04-24)
```
It must **not** appear again in the custom fields block below (i.e. `customfield_10020` should be suppressed from the raw custom fields since it is surfaced in the dedicated Sprint line).

**2. Transitions block format.** In the same output, the Transitions block must use indented pipe-delimited lines with **no** header row:
```
Transitions:
  91 | Ready | Ready for Development
  161 | Start | In Development
  11 | Backlog | Backlog
  21 | Ready for Development | Ready for Development
  31 | In Development | In Development
  41 | In Code Review | In Code Review
  51 | Ready for QA | Ready for QA
  61 | Ready for Deployment | Ready for Deployment
  71 | Deployed | Deployed
  81 | Canceled | Canceled
```
(The `ID | NAME` header row must be absent.)

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Resolves sprint from `customfield_10020` when the typed `sprint` field is nil (Jira returns sprint as a custom field in the REST API)
- Suppresses `customfield_10020` from the custom fields block when Sprint is populated (both extended and projection paths)
- Replaces transitions table (with header row) with indented pipe-delimited lines: `ID | NAME | TO_STATUS`
- Updates projection `Transitions` format to include `To.Name`

Closes #303

### CODE REVIEW

**Comment by rianjs** [comment]
## TDD Assessment

Overall coverage is solid. The PR ships meaningful tests alongside its implementation changes. A few gaps worth flagging.

---

### What's well-covered

**`resolveSprintFromCustomField` (types.go)**
Four targeted tests cover the core polymorphism: array (takes last element), single-object fallback, empty array → nil, and `null` → nil. The typed-field-wins precedence test correctly verifies the `if f.Sprint == nil` guard.

**Presenter transitions format change**
Both the `Extended` and `Projection` rendering paths are tested, and the integration test in `get_test.go` confirms the header row (`ID | NAME`) is gone. The three-column format (`id | name | to-name`) is verified at all layers.

**`customfield_10020` suppression**
Both `PresentDetailExtended` and `PresentDetailProjection` are tested to confirm the raw custom field is hidden when `Sprint` is populated, and that other custom fields still render.

---

### Gaps

**`resolveSprintFromCustomField` — malformed/unexpected JSON**
The function has a silent double-fallback (`[]Sprint` → `Sprint` → `nil`). There is no test for a value that is neither a Sprint array nor a Sprint object (e.g., a bare string `"some legacy value"` or a number). In the wild, Jira occasionally serialises `customfield_10020` as a string when the issue was never in a sprint. Without a test, we can't confirm the function returns `nil` gracefully rather than misinterpreting the value.

**`resolveSprintFromCustomField` — single-Sprint object path**
The array path is tested; the single-object path (`var s Sprint`) is not. If the fallback unmarshal into `Sprint` succeeds but the struct is zero-valued (missing required fields), the caller gets a non-nil pointer with an empty Sprint. A test with a well-formed single-object payload would close this.

**Presenter — Sprint with dates vs. without**
`TestIssuePresenter_PresentDetailExtended_SprintResolved` uses a Sprint without dates. The integration test (`TestRunGet_Extended_SprintFromCustomField`) asserts `Sprint: MON Sprint 70 (id: 125, active, 2026-04-10 → 2026-04-24)`, confirming the date range renders. A unit-level presenter test for the date-range branch (and for a Sprint where dates are absent) would lock down the formatting contract without needing a full HTTP server.

**`customfield_10020` suppression in `PresentDetailProjection`**
The projection test exercises the suppression for the `Extended` path. The projection path also has the same `continue` guard (line ~459 of issue.go), but there is no test asserting that `customfield_10020` is absent from the projection output when Sprint is set. Minor, but the symmetry would be good to have.

---

### Minor flags

- The `issueJSON` fixture in `TestRunGet_Extended_SprintFromCustomField` hardcodes a real-looking key (`MON-4970`) and real sprint IDs. This is fine functionally but can confuse future readers about whether the test is load-bearing on specific IDs.
- `TestIssueFields_SprintFromCustomField10020_TypedFieldWins` doesn't assert that `f.Sprint.ID == 99` (only name). Both checks would make the precedence contract unambiguous.

---

**Verdict**: Approve with the single-object fallback and malformed-input gaps noted. Neither is a blocker, but both are easy wins given the existing test structure.

File: tools/jtk/api/types.go [hunk 1]
```diff
@@ -79,6 +79,30 @@ func (f *IssueFields) UnmarshalJSON(data []byte) error {
...
+func resolveSprintFromCustomField(raw json.RawMessage) *Sprint {
+	if len(raw) == 0 || string(raw) == "null" {
+		return nil
+	}
+	var arr []Sprint
+	if err := json.Unmarshal(raw, &arr); err == nil {
+		if len(arr) == 0 {
+			return nil
+		}
+		return &arr[len(arr)-1]
```
Comment (monit-reviewer) [inline tools/jtk/api/types.go:100]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): resolveSprintFromCustomField always returns the last element of the array without checking sprint state. This implicitly assumes Jira returns sprints ordered oldest-to-newest with the active sprint last. If Jira ever reverses order or returns multiple active sprints, the wrong sprint would be selected silently. Consider preferring the element with state == 'active' before falling back to the last element.

*Reply to this thread when addressed.*

File: tools/jtk/api/types_test.go [hunk 2]
```diff
@@ -722,3 +722,86 @@ func TestIssue_RoundTrip_WithCustomFields(t *testing.T) {
...
+	t.Parallel()
+	input := `{
+		"summary": "Test",
+		"customfield_10020": {"id": 125, "name": "Single Sprint", "state": "active"}
+	}`
+	var f IssueFields
+	err := json.Unmarshal([]byte(input), &f)
+	testutil.RequireNoError(t, err)
+	if f.Sprint == nil {
+		t.Fatal("Sprint should be resolved from single-object customfield_10020")
```
Comment (monit-reviewer) [inline tools/jtk/api/types_test.go:756]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): No test covers the case where customfield_10020 is an array containing only closed or future sprints (no active sprint). The current behavior returns the last element regardless of state, but this is untested — a reader cannot tell if returning a closed sprint is intentional or an oversight.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/issue_test.go [hunk 3]
```diff
@@ -449,6 +449,126 @@ func TestIssuePresenter_PresentTypeAlreadyCurrent(t *testing.T) {
...
+	t.Parallel()
+	issue := &api.Issue{
+		Key:    "MON-123",
+		Fields: api.IssueFields{Summary: "Test"},
+	}
+	dctx := &DetailContext{
+		Transitions: []api.Transition{
+			{ID: "11", Name: "Backlog", To: api.Status{Name: "Backlog"}},
+		},
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/issue_test.go:524]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): The type assertion `model.Sections[0].(*present.DetailSection)` has no comma-ok check. If PresentDetailProjection ever changes its section layout, this panics instead of producing a clean test failure. Use the two-value form and call t.Fatalf on failure.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/issue_test.go [hunk 4]
```diff
@@ -449,6 +449,126 @@ func TestIssuePresenter_PresentTypeAlreadyCurrent(t *testing.T) {
...
+
+	model := IssuePresenter{}.PresentDetailProjection(issue, "", false, dctx)
+	detail := model.Sections[0].(*present.DetailSection)
+
+	for _, f := range detail.Fields {
+		if f.Label == "Custom_Fields" {
+			if strings.Contains(f.Value, "customfield_10020") {
+				t.Error("customfield_10020 should be suppressed in projection when Sprint is populated")
+			}
+			if !strings.Contains(f.Value, "customfield_10035") {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/issue_test.go:563]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): Same unguarded type assertion `model.Sections[0].(*present.DetailSection)` in TestIssuePresenter_PresentDetailProjection_CustomFieldSuppression. Same risk: panic on layout change rather than clean test failure. Use the two-value form and call t.Fatalf on failure.

*Reply to this thread when addressed.*

File: tools/jtk/internal/present/issue.go [hunk 5]
```diff
@@ -476,7 +477,7 @@ func (IssuePresenter) PresentDetailProjection(issue *api.Issue, _ string, fullte
 	if dctx != nil && len(dctx.Transitions) > 0 {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/issue.go:477]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The projection transitions format changes from `ID:Name` to `ID:Name:ToName`, which is a breaking change for consumers of the machine-readable projection output. Any script parsing `ID:Name` pairs will silently receive an unexpected third segment. The PR description does not call this out as a breaking change.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `a611f89`

> **Approved** with 5 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 2 |
| harness-engineering:harness-enforcement-reviewer | 1 |
| harness-engineering:harness-knowledge-reviewer | 2 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/types.go:100`
> resolveSprintFromCustomField always returns the last element of the array without checking sprint state. This implicitly assumes Jira returns sprints ordered oldest-to-newest with the active sprint last. If Jira ever reverses order or returns multiple active sprints, the wrong sprint would be selected silently. Consider preferring the element with state == 'active' before falling back to the last element.

**💡 Suggestion** - `tools/jtk/internal/present/issue.go:477`
> The projection transitions format changes from `ID:Name` to `ID:Name:ToName`, which is a breaking change for consumers of the machine-readable projection output. Any script parsing `ID:Name` pairs will silently receive an unexpected third segment. The PR description does not call this out as a breaking change.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/types_test.go:756`
> No test covers the case where customfield_10020 is an array containing only closed or future sprints (no active sprint). The current behavior returns the last element regardless of state, but this is untested — a reader cannot tell if returning a closed sprint is intentional or an oversight.

</details>

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/issue_test.go:524`
> The type assertion `model.Sections[0].(*present.DetailSection)` has no comma-ok check. If PresentDetailProjection ever changes its section layout, this panics instead of producing a clean test failure. Use the two-value form and call t.Fatalf on failure.

**💡 Suggestion** - `tools/jtk/internal/present/issue_test.go:563`
> Same unguarded type assertion `model.Sections[0].(*present.DetailSection)` in TestIssuePresenter_PresentDetailProjection_CustomFieldSuppression. Same risk: panic on layout change rather than clean test failure. Use the two-value form and call t.Fatalf on failure.

</details>

*4 info-level observations excluded. Run with `--verbose` to include.*

*1 PR discussion thread considered.*

---
<details>
<summary>Completed in 1m 46s | $0.65</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 1m 46s wall · 4m 50s compute (Reviewers: 1m 08s · Synthesis: 35s) |
| Cost | $0.65 |
| Tokens | 268.9k in / 17.6k out |
| Turns | 18 |

</details>

File: tools/jtk/api/types.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/api/types.go:100]: Acknowledged — Jira returns the active sprint last in the array. If that changes, we can add state-aware selection.

File: tools/jtk/api/types_test.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/api/types_test.go:756]: The last-element heuristic matches Jira's API ordering. All-closed is a valid edge case but low risk.

File: tools/jtk/internal/present/issue_test.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/present/issue_test.go:524]: Test intentionally panics on wrong section type — that's how we catch contract violations.

File: tools/jtk/internal/present/issue_test.go (hunk 4 above)
Comment (rianjs) [reply tools/jtk/internal/present/issue_test.go:563]: Same as above — intentional panic-on-wrong-type test pattern.

File: tools/jtk/internal/present/issue.go (hunk 5 above)
Comment (rianjs) [reply tools/jtk/internal/present/issue.go:477]: This is intentional — the projection format was missing To.Name which is needed for the output spec.
